pyaedt/modules/solutions.py

```diff
@@ -1797,6 +1801,112 @@ def plotsettings(self):
+            List with [x,y,z] coordinates of a point or list of lists of points or
+            dictionary with keys containing point names and for each key the point
+            coordinates [x,y,z].
+        filename : str, optional
+            Full path or relative path with filename.
+            Default is ``None`` in which case no file is exported.
+        visibility : bool, optional
+            Whether to keep the markers visible in the UI. Default is ``False``.
+
+        Returns
+        -------
+        dict or pd.DataFrame
+            Dict containing 5 keys: point names, x,y,z coordinates and the quantity probed.
+            Each key is associated with a list with the same length of the argument points.
+            If pandas is installed, the output is a pandas DataFrame with point names as
+            index and coordinates and quantity as columns.
+        """
+        self.oField.ClearAllMarkers()
+
+        # Clean inputs
+        if isinstance(points, dict):
+            points_name = list(points.keys())
+            points_value = [points[pt_name] for pt_name in points_name]
+        elif isinstance(points, list):
+            points_name = None
+            if not isinstance(points[0], list):
+                points_value = [points]
+            else:
+                points_value = points
+        else:
+            raise AttributeError("``points`` argument is invalid.")
+        if filename is not None:
+            if not os.path.isdir(os.path.dirname(filename)):
+                raise AttributeError("Specified path ({}) does not exist".format(filename))
+
+        # Create markers
+        u = self._postprocessor._app.modeler.model_units
+        added_points_name = []
+        for pt_name_idx, pt in enumerate(points_value):
+            try:
+                pt = [c if isinstance(c, str) else "{}{}".format(c, u) for c in pt]
+                self.oField.AddMarkerToPlot(pt, self.name)
+                if points_name is not None:
+                    added_points_name.append(points_name[pt_name_idx])
+            except (GrpcApiError, SystemExit) as e:  # pragma: no cover
+                self._postprocessor.logger.error(
+                    "Point {} not added. Check if it lies inside the plot.".format(str(pt))
+                )
+                raise e
+
+        # Export data
+        with tempfile.NamedTemporaryFile(mode="w+", delete=False, suffix=".csv") as temp_file:
+            temp_file.close()
+            self.oField.ExportMarkerTable(temp_file.name)
+            with open_file(temp_file.name, "r") as f:
+                reader = csv.DictReader(f)
+                out_dict = defaultdict(list)
+                for row in reader:
+                    for key in row.keys():
+                        if key == "Name":
+                            val = row[key]
+                        else:
+                            val = float(row[key].lstrip())
+                        out_dict[key.lstrip()].append(val)
```

Comment:
It is a bit awkward to using a context manager to open a file and closing it right away.
Could you create the temp_file through classical variable assignment, call `ExportMarkerTable` afterward and then use the context manager if needed ?

```python
temp_file = tempfile..NamedTemporaryFile(mode="w+", delete=False, suffix=".csv")
temp_file.close()
self.oField.ExportMarkerTable(temp_file.name)
...
```